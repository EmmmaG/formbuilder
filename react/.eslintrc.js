module.exports = {
    globals: {},
    env: {
        commonjs: true,
        es6: true,
    },
    parser: '@typescript-eslint/parser',
    plugins: ['@typescript-eslint'],
    parserOptions: {
        ecmaVersion: 2018,
        ecmaFeatures: {
            globalReturn: false,
        },
        sourceType: 'module',
    },
}
