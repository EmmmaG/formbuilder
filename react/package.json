{
  "name": "hcard-builder",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@material-ui/core": "^3.9.2",
    "@material-ui/icons": "^3.0.2",
    "@types/react-dom": "^16.8.3",
    "react": "^16.8.4",
    "react-dom": "^16.8.4",
    "react-scripts": "2.1.8",
    "typescript": "^3.3.4000"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "jest",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "browserslist": [
    ">0.2%",
    "not dead",
    "not ie <= 11",
    "not op_mini all"
  ],
  "description": "This project was to conver a fixed format file into csv",
  "main": "index.js",
  "devDependencies": {
    "@types/jest": "^23.1.5",
    "@typescript-eslint/eslint-plugin": "^1.5.0",
    "@typescript-eslint/parser": "^1.5.0",
    "@types/enzyme": "^3.1.11",
    "@types/enzyme-adapter-react-16": "^1.0.1",
    "babel-core": "^6.26.3",
    "babel-loader": "^7.1.5",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-es2015-modules-commonjs": "^6.26.2",
    "babel-plugin-transform-object-rest-spread": "^6.23.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-preset-env": "^1.7.0",
    "babel-preset-react": "^6.24.1",
    "babel-runtime": "^6.25.0",
    "enzyme": "^3.2.0",
    "enzyme-adapter-react-16": "^1.1.1",
    "enzyme-to-json": "^3.3.4",
    "jest": "^24.1.0",
    "jest-enzyme": "^7.0.1",
    "jest-fetch-mock": "^1.6.5",
    "ts-jest": "^22.4.6"
  },
  "author": "",
  "license": "ISC"
}
