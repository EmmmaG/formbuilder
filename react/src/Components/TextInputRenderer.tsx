import * as React from 'react'
import { withStyles, WithStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import TextField from '@material-ui/core/TextField'
import Button from '@material-ui/core/Button'
import { TextInput } from '../Components/TextInputBuilder'

const styles = () => ({
    container: {
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        margin: '20px',
    },
    deleteButton: {
        height: '40px',
    }
})

export interface Props extends WithStyles<typeof styles> {
    field: TextInput
    onDelete: (id: number) => void
}

export interface State {
    value: string
}

export class TextInputRenderer extends React.Component<Props, State> {
    public constructor(props: Props) {
        super(props)

        this.state = {
            value: this.props.field.defaultValue || '',
        }
    }

    private onDelete = () => {
        this.props.onDelete(this.props.field.id!)
    }

    private onTextFieldChange = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>) => {
        this.setState({ value: e.target.value })
    }

	public render() {
        const { classes, field } = this.props
        const { value } = this.state

		return (
            <div className={classes.container}>
                <div>
                    <Typography>{field.question}</Typography>
                    <TextField
                        key={field.id}
                        label={field.label}
                        placeholder={field.placeholder}
                        value={value}
                        onChange={(e) => this.onTextFieldChange(e)}
                        margin='normal'
                    />
                </div>
                <Button
                    variant='outlined'
                    color='secondary'
                    onClick={this.onDelete}
                    className={classes.deleteButton}
                >
                    Delete
                </Button>
            </div>
		)
	}
}

export default withStyles(styles)(TextInputRenderer)
