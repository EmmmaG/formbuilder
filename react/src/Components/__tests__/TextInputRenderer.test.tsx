import * as React from 'react'
import { shallow } from 'enzyme'
import idObj from 'identity-obj-proxy'

import TextInputRenderer, { Props } from '../TextInputRenderer'

const props: Props = {
    field: {
        id: 1,
        type: 'textInput',
        question: 'which day you like best?',
        placeholder: 'day',
    },
    onDelete: jest.fn(),
	classes: idObj,
}

describe('TextInputRenderer', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<TextInputRenderer {...props} />).dive()
		expect(wrapper).toMatchSnapshot()
	})
})
