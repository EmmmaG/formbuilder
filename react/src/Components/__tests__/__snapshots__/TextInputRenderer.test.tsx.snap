// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInputRenderer renders correctly 1`] = `
<div
  className="TextInputRenderer-container-1"
>
  <div>
    <WithStyles(Typography)>
      which day you like best?
    </WithStyles(Typography)>
    <TextField
      key="1"
      margin="normal"
      onChange={[Function]}
      placeholder="day"
      required={false}
      select={false}
      value=""
      variant="standard"
    />
  </div>
  <WithStyles(Button)
    className="TextInputRenderer-deleteButton-2"
    color="secondary"
    onClick={[Function]}
    variant="outlined"
  >
    Delete
  </WithStyles(Button)>
</div>
`;
